### PR TITLE
Allow optional namespaces value when calling kubectl apply.

### PR DESCRIPTION
**Pull request by olymk2**
I had a stab at adding namespace support to kube ctl, Implemented in a way it should not effect existing functionality.

Still quite new to clojure so if I am doing anything glaringly bad let me know :)

### CODE REVIEW

**Comment by olymk2** [comment]
@brosenan are you accepting pr's ? just updated this to allow for kubeconfig param ass well as namespace.

**Comment by brosenan** [comment]
Sorry for the delay in my response.

The build for this PR is failing. I think the root cause is out-of-date dependencies. I did a quick fix, setting the jdk version to openjdk9 to make head build again.

Please rebase this PR and I'll accept it.

**Comment by brosenan** [comment]
Please add tests + documentation for the new parameters. LMK if you need any assistance.

https://github.com/brosenan/lambda-kube/blob/7c0c5093108516b2ea7af36cae9050a5bf481d8c/test/lambdakube/core_test.clj#L1309

**Comment by olymk2** [comment]
@brosenan I had a little stab at adding a test and some text to the readme, need to look into the fact library at some point its layout is a bit foreign to me.

File: test/lambdakube/core_test.clj [hunk 1]
```diff
@@ -1307,6 +1307,7 @@ spec:
 ")
 
 ;; `kube-apply` takes a string constructed by `to-yaml` and a `.yaml`
+;; optionally supply namespace and kube config parameters.
```
Comment (brosenan) [inline test/lambdakube/core_test.clj]: This line seems to be out of place. It is in the middle of a sentence...

I would move it to just before line 1323, to describe the new test you created.

**Review by brosenan — CHANGES_REQUESTED** [review]
Please run `./create-docs.sh ` to update the `.md` files. They should be part of the PR.

Thanks,
Boaz.

File: test/lambdakube/core_test.clj (hunk 1 above)
Comment (olymk2) [reply test/lambdakube/core_test.clj]: okay made those changes.

**Review by brosenan — CHANGES_REQUESTED** [review]
Hi Oliver, thanks for your effort, and thanks for baring with me.

I just realized you made manual changes to `core.md`, and parallel changes to `core_test.clj`.

I'm using an approach in which the documentation (`.md` files) are directly derived from the tests (`_test.clj` files). Comments are taken to be free text and Clojure code is taken as code fragments.

This process is done automatically using the `./create-docs.sh` script.

For this reason, the comments in `core_test.clj` are supposed to document the _implementation,_ not the _test_. The line you wrote:
```
optionally supply namespace and kube config parameters.
```
would make a great comment for the test you have added (instead of the documentation you added in `core_test.clj`, which describes the test and not the feature you implemented.

Once you made the change, run `./create_docs.sh` to generate the docs. If you have trouble running it on your machine (e.g., you're using Windows and have no sh/awk), let me know and I'll merge the code and run the script for you.

Thanks again,
Boaz.

**Comment by olymk2** [comment]
okay will change that when i get 2 secs, never seen docs generated from tests before, I had assumed incorrectly they where comment related to tests, does that ean you cant comment your tests ?

**Comment by olymk2** [comment]
@brosenan okay modified, I may have run the create docs script before modifying the tests, I had not realized the md files was generated from the tests comments, thanks for guiding me I may have a few more pr's going forward so this is helpful :)

**Comment by brosenan** [comment]
> okay will change that when i get 2 secs, never seen docs generated from tests before, I had assumed incorrectly they where comment related to tests, does that ean you cant comment your tests ?

If the test is not self explanatory, you can add comments inside the `(fact)` form.

**Pull request merged**